packages/js/src/generators/library/library.ts

```diff
@@ -540,32 +538,39 @@ function normalizeOptions(
     options.linter = Linter.EsLint;
   }
 
-  const projectName = options.rootProject
-    ? name
-    : projectDirectory.replace(new RegExp('/', 'g'), '-');
+  const {
+    projectName,
+    names: projectNames,
+    projectDirectory,
+    importPath,
+  } = await determineProjectNamesAndDirectories(tree, {
+    name: options.name,
+    projectType: 'library',
+    directory: options.directory,
+    importPath: options.importPath,
+    nameDirectoryFormat: options.nameDirectoryFormat,
+    rootProject: options.rootProject,
+  });
+  options.rootProject = projectDirectory === '.';
```

Comment:
Not exactly. If `rootProject` is provided, is considered for both formats. It's only relevant if the directory was not provided though.

If we remove it when `derived` is removed, we lose the ability to create what we call a root project. We decided to use the project name as the directory if the directory is not provided for the new way. So, if `rootProject` doesn't exist, you can't create a project in the root because you'll always get a folder with the project name.